Svc/FileDownlink/FileDownlink.hpp

```diff
@@ -210,6 +209,13 @@ namespace Svc {
           FileDownlink *const fileDownlink;
 
       };
+      //! Enumeration for packet types
+      //! Each type has a buffer to store it.
+      enum PacketType {
+          FILE_PACKET,
+          CANCEL_PACKET,
+          COUNT_PACKET_TYPE
```

Comment:
Is this the best way to get the number of elements in an enum? I'm guessing you did this because there isn't a better way, but I just have to ask 😂 . It took me a second to figure out how this worked.

Reply:
We have a pattern of adding an extra type like that on enums of this style for assertion purposes.  This is done all over F´.  I don't know of a better way given the older versions of C++ we support.